bittensor/_neuron/text/core_validator/__init__.py

```diff
@@ -606,24 +610,34 @@ def run_epoch( self ):
         self.prometheus_gauges.labels("dividends").set( self.metagraph.dividends[self.uid] )
         self.prometheus_gauges.labels("emission").set( self.metagraph.emission[self.uid] )
 
+        # Iterate epochs.
+        self.epoch += 1
+
     def metagraph_sync(self):
         r""" Syncing metagraph together with other metagraph-size related objects
         """
-        old_hotkeys = self.neuron_hotkeys if self.neuron_hotkeys else self.metagraph.hotkeys
-        self.metagraph.sync()
+        old_hotkeys = self.neuron_hotkeys + [] if self.neuron_hotkeys else self.metagraph.hotkeys
+        self.metagraph.sync(cached=self.config.neuron.metagraph_cached or self.subtensor.network != 'local')
         self.neuron_hotkeys = self.metagraph.hotkeys
 
+        assert len(old_hotkeys) == len(self.neuron_hotkeys), 'metagraph hotkeys length do not match'
```

Comment:
This may break the validators when one day we decide to increase the number of peers from 4096 but we forget we have this constrain here xd